engine/src/events/callback/event_function_handler.h

```diff
@@ -4,34 +4,37 @@
 
 #pragma once
 
-#include "handler.h"
+#include "event_handler.h"
 
 namespace Bald {
-
+    class Event;
     /**
      * @class FunctionHandler
      * @brief Synchronous function wrapper implementation
      */
 
-    class FunctionHandler : public Handler {
+    template <typename E>
+    class EventFunctionHandler : public EventHandler<E> {
     public:
 
         /**
         * @fn                   FunctionHandler
         * @brief                Constructor
-        * @param [F&&]          fun -> Function which will be wrapped
+        * @param [F&&]          fun -> Function which will be wrapped *HAS TO* take as first parameter an const reference to Event
```

Comment:
fun -> Function which will be wrapped. This method MUST take const reference to Event as first parameter.